pymc/model_graph.py

```diff
@@ -100,19 +100,22 @@ def make_compute_graph(
             input_map[var_name] = input_map[var_name].union(parent_name)
 
             while True:
-                if hasattr(var.tag, "observations"):
-                    obs_node = var.tag.observations
-                    obs_name = obs_node.name
-                    if obs_name and obs_name != var_name:
-                        input_map[var_name] = input_map[var_name].difference({obs_name})
-                        input_map[obs_name] = input_map[obs_name].union({var_name})
+                try:
+                    if hasattr(var.tag, "observations"):
+                        obs_node = var.tag.observations
+                        obs_name = obs_node.name
+                        if obs_name and obs_name != var_name:
+                            input_map[var_name] = input_map[var_name].difference({obs_name})
+                            input_map[obs_name] = input_map[obs_name].union({var_name})
+                            break
+                        elif isinstance(obs_node.owner.op, Elemwise) and isinstance(
+                            obs_node.owner.inputs[0].owner.op.scalar_op, Cast
+                        ):
+                            # go to the beginning of the loop
+                            var.tag.observations = obs_node.owner.inputs[0].owner.inputs[0]
+                    else:
                         break
-                    elif isinstance(obs_node.owner.op, Elemwise) and isinstance(
-                        obs_node.owner.inputs[0].owner.op.scalar_op, Cast
-                    ):
-                        # go to the beginning of the loop
-                        var.tag.observations = obs_node.owner.inputs[0].owner.inputs[0]
-                else:
+                except AttributeError:
```

Comment:
Where is the `AttributeError` coming from?
Couldn't it be solved with a `hasattr` in the same way like above?

Reply:
Great point, I believe that it should. For some data containers, `obs_node` would node have an `owner` attribute. I am checking the tests locally and pushing if all passes.